### PR TITLE
feat: Allow Mui.Modal & Mui.Dialog props

### PR DESCRIPTION
**Pull request by Crash--**
Allow the use of https://v3.material-ui.com/api/dialog/#props & https://v3.material-ui.com/api/modal/#props

### CODE REVIEW

File: react/CozyDialogs/Dialog.jsx [hunk 1]
```diff
@@ -35,6 +36,7 @@ const Dialog = ({
       fullScreen={isFullscreen}
       classes={{ paper: paperClassName }}
       aria-labelledby={`modal-title-${id}`}
+      {...otherProps}
```
Comment (ptbrowne) [inline react/CozyDialogs/Dialog.jsx:39]: classes is overriden here ? I am not sure that it is what's expected by the user. I think otherProps.classes should be merged.

Comment (ptbrowne) [reply react/CozyDialogs/Dialog.jsx:39]: Same thing for TransitionProps

Comment (Crash--) [reply react/CozyDialogs/Dialog.jsx:39]: Nice catch 👍

File: react/CozyDialogs/Readme.md [hunk 2]
```diff
@@ -15,6 +15,7 @@ Pre-built modals ready to be directly used in applications, based on MUI Dialog.
 * content : `<node>` Content of the modal
 * actions : `<node>` Actions of the modal
 * actionsLayout : `<string>` Can be "row" or "column"
+* ...Mui.ModalProps & ...Mui.DialogProps
```
Comment (ptbrowne) [inline react/CozyDialogs/Readme.md:18]: Can you add a link to the respective components ? I think a phrase would be easier to read.

File: react/CozyDialogs/Dialog.jsx (hunk 1 above)
Comment (ptbrowne) [reply react/CozyDialogs/Dialog.jsx:39]: Could you modify the custom Dialog example with the utilisation of these features ?

Comment (ptbrowne) [reply react/CozyDialogs/Dialog.jsx:39]: I've done it in https://github.com/cozy/cozy-ui/pull/1640

File: react/CozyDialogs/Readme.md (hunk 2 above)
Comment (ptbrowne) [reply react/CozyDialogs/Readme.md:18]: I've done it in https://github.com/cozy/cozy-ui/pull/1640

**Pull request closed without merging**